src/lib/dlt_user.c

```diff
@@ -2917,9 +2917,20 @@ void dlt_user_trace_network_segmented_thread(void *unused)
     prctl(PR_SET_NAME, "dlt_segmented", 0, 0, 0);
 #endif
 
+    int count = 0;
     s_segmented_data *data;
 
     while (1) {
+        // Normally reattaching to dlt-daemon is triggered when app tries to output log.
+        // Even if the app outputs log only before starting up of the dlt-daemon,
+        // this retry loop helps reattaching.
+        while ((dlt_user.dlt_log_handle < 0) && (count < DLT_USER_REATTACH_MAX_COUNT))
+        {
+            dlt_user_log_reattach_to_daemon();
```

Comment:
I would propose to handle it somewhere else like in dlt_receiver thread because this situation is not only for network trace but also normal logging. The thing is we need to flush all the data in internal buffer to daemon after daemon started which is only done when new log message comes from application. Let us work on it a bit.

Reply:
Yes, I also dlt_receiver thread can also do same thing. However I wanted to avoid adding new route for output log. If we do same thing from dlt_receiver thread, there will be 3 routes.
1. Output from Main thread (Thread of application)
2. Output from Network segmented thread
3. Output from Receiver thread
I thought this confuses us and it may have risk.